embrace-android-api/src/main/kotlin/io/embrace/android/embracesdk/internal/capture/activity/OpenEvents.kt

```diff
@@ -0,0 +1,62 @@
+package io.embrace.android.embracesdk.internal.capture.activity
+
+/**
+ * The relevant stages in the lifecycle of Activities pertaining to observing the performance of their loading
+ */
+public interface OpenEvents {
+
+    /**
+     * When a previously in-progress Activity Open trace should be abandoned, and that the component managing
+     * the trace recording should prepare itself to start tracing the opening of a new Activity instance.
+     */
+    public fun resetTrace(instanceId: Int, activityName: String, timestampMs: Long)
+
+    /**
+     * When the app is no longer in a state where it is trying to open up a new Activity
+     */
+    public fun hibernate(instanceId: Int, activityName: String, timestampMs: Long)
```

Comment:
This is when one activity's `onStop()` is called without another's `onStart()` is called. It is meant to denote the app going into the background so another activity has not been started.

We keep track of this in order to properly gauge the start time of the next activity, as I wanted to account for the time between `onPause` and `onCreate` between two activities and have it as part of the open time for the latter if we are doing a foreground transition. Obviously we don't want to track this if the app was previously backgrounded.

Reply:
Got it. So this is equivalent to `ProcessStateService#onBackground`

Reply:
Yeah, except I think this will fire for an activity even if it opened another activity, whereas I think the `LifecycleEventObserver` only fires `onStop` if it doesn't detect that another activity is being opened?

Hmm. Makes me think whether I should hook into this instead of the activity itself to record these traces....